relay-server/src/actors/upstream.rs

```diff
@@ -661,6 +772,14 @@ impl Handler<PumpHttpMessageQueue> for UpstreamRelay {
             return;
         }
 
+        // During network outages no message will be sent
+        if self.is_network_outage() {
+            // initiate a network checking loop if not already started
+            self.check_network_status(ctx);
+            return;
+        }
```

Comment:
I believe you can get rid of this check and consolidate all network error handling into just the response handler if you do the following:

- `is_network_outage` is removed
- `is_ready` also checks for network outages like the following:
- a network outage is defined as `network_outage_backoff.is_started()`
- `handle_http_response` is the only place where you deal with the network error, with the exact same code as in master.

Do you think that would work? I would favor this, because it reduces the places where we have to check for errors.